<a href="https://colab.research.google.com/github/Tej-v/TripAdvisor-Dataset/blob/main/Trip_Advisor_Dataset.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

In [ ]:
from google.colab import files
uploaded = files.upload()

Saving hotel_reviews.csv to hotel_reviews.csv


In [ ]:
import pandas as pd

In [ ]:
#Load the dataset (Trip Advisor)

In [ ]:
df = pd.read_csv('hotel_reviews.csv')

In [ ]:
print(df.head()) #display first few rows of the datset

   Index             Name                  Area Review_Date  \
0      0  Hotel The Pearl  Paharganj, New Delhi      Jul-23   
1      1  Hotel The Pearl  Paharganj, New Delhi      Aug-23   
2      2  Hotel The Pearl  Paharganj, New Delhi      Aug-23   
3      3  Hotel The Pearl  Paharganj, New Delhi      Aug-23   
4      4  Hotel The Pearl  Paharganj, New Delhi      Aug-23   

                            Rating_attribute  Rating(Out of 10)  \
0                 Best budget friendly hotel                9.0   
1                              Amazing place                9.0   
2               Overall good stay. Economic.                9.0   
3                                     Lovely                9.0   
4  Great hotel Great staff and great staying                9.0   

                                         Review_Text  
0  Hotel the pearl is perfect place to stay in De...  
1  Location of the hotel is perfect. The hotel is...  
2                             Location, Indian food. 

In [ ]:
#Inspecting the dataset

Goal = To understand the structure and contents of the dataset to identify potential issues or areas for cleaning.

In [ ]:
df.info() #Cheching the coloumns and their Data Types

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 7001 entries, 0 to 7000
Data columns (total 7 columns):
 #   Column             Non-Null Count  Dtype  
---  ------             --------------  -----  
 0   Index              7001 non-null   int64  
 1   Name               7001 non-null   object 
 2   Area               7001 non-null   object 
 3   Review_Date        7001 non-null   object 
 4   Rating_attribute   7001 non-null   object 
 5   Rating(Out of 10)  7001 non-null   float64
 6   Review_Text        6994 non-null   object 
dtypes: float64(1), int64(1), object(5)
memory usage: 383.0+ KB


This tells us:

There are 7,001 rows and 7 columns in the dataset.
All columns except Review_Text have no missing values.
The Review_Text column has 7 missing values (since it has 6,994 non-null entries).
Data types: The Index and Rating(Out of 10) columns are numerical (int64 and float64), while the rest are object types (which likely means strings).

In [ ]:
df.isnull().sum() #Checking for any missing values

Index                0
Name                 0
Area                 0
Review_Date          0
Rating_attribute     0
Rating(Out of 10)    0
Review_Text          7
dtype: int64

This confirms that:

Only the Review_Text column has missing data, with 7 missing entries.

In [ ]:
df.describe() #Checking Summary Statistics for numerical columns

Index  Rating(Out of 10)
count  7001.00000        7001.000000
mean   3500.00000           7.030981
std    2021.15895           2.882846
min       0.00000           1.000000
25%    1750.00000           6.000000
50%    3500.00000           8.000000
75%    5250.00000           9.000000
max    7000.00000          10.000000

This shows basic summary statistics for the numerical columns (Index and Rating(Out of 10)):

The mean rating is about 7.03, which suggests that on average, hotels receive fairly positive reviews.
The minimum rating is 1, and the maximum is 10.
The 25th percentile (first quartile) rating is 6, the median (50th percentile) rating is 8, and the 75th percentile is 9.
This gives us a general sense of the distribution of hotel ratings in the dataset.

In [ ]:
#Handling Missing Data

In [ ]:
df_cleaned = df.dropna(subset=['Review_Text'])

In [ ]:
#Data Pre-Proccesing

In [ ]:
df_cleaned = df_cleaned.drop(columns= ['Index']) #dropping Index column

In [ ]:
print(df_cleaned.head())

              Name                  Area Review_Date  \
0  Hotel The Pearl  Paharganj, New Delhi      Jul-23   
1  Hotel The Pearl  Paharganj, New Delhi      Aug-23   
2  Hotel The Pearl  Paharganj, New Delhi      Aug-23   
3  Hotel The Pearl  Paharganj, New Delhi      Aug-23   
4  Hotel The Pearl  Paharganj, New Delhi      Aug-23   

                            Rating_attribute  Rating(Out of 10)  \
0                 Best budget friendly hotel                9.0   
1                              Amazing place                9.0   
2               Overall good stay. Economic.                9.0   
3                                     Lovely                9.0   
4  Great hotel Great staff and great staying                9.0   

                                         Review_Text  
0  Hotel the pearl is perfect place to stay in De...  
1  Location of the hotel is perfect. The hotel is...  
2                             Location, Indian food.  
3  The location and the hotel itself is 

In [ ]:
#Formatting Date Column to ensure it follows the Date-Time format

In [ ]:
df_cleaned['Review_Date'] = pd.to_datetime(df_cleaned['Review_Date'], format= '%b-%y')

In [ ]:
print(df_cleaned.head())

              Name                  Area Review_Date  \
0  Hotel The Pearl  Paharganj, New Delhi  2023-07-01   
1  Hotel The Pearl  Paharganj, New Delhi  2023-08-01   
2  Hotel The Pearl  Paharganj, New Delhi  2023-08-01   
3  Hotel The Pearl  Paharganj, New Delhi  2023-08-01   
4  Hotel The Pearl  Paharganj, New Delhi  2023-08-01   

                            Rating_attribute  Rating(Out of 10)  \
0                 Best budget friendly hotel                9.0   
1                              Amazing place                9.0   
2               Overall good stay. Economic.                9.0   
3                                     Lovely                9.0   
4  Great hotel Great staff and great staying                9.0   

                                         Review_Text  
0  Hotel the pearl is perfect place to stay in De...  
1  Location of the hotel is perfect. The hotel is...  
2                             Location, Indian food.  
3  The location and the hotel itself is 

In [ ]:
#Checking for duplicates

In [ ]:
duplicates = df_cleaned.duplicated().sum()

In [ ]:
print(f'Number of duplicate rows: {duplicates}')

Number of duplicate rows: 2314


In [ ]:
#Removing Duplicates

In [ ]:
df_cleaned = df_cleaned.drop_duplicates()

In [ ]:
#Checking the shape of the dataset after removinf the duplicates

In [ ]:
print(df_cleaned.shape)

(4680, 6)
